rs/index/src/ivf/files/invalidated_ids.rs

```diff
@@ -131,6 +141,57 @@ impl InvalidatedIdsStorage {
 
         Ok(())
     }
+
+    pub fn iter(&mut self) -> InvalidatedIdsIterator {
+        InvalidatedIdsIterator {
+            files: (0..self.files.len())
+                .filter_map(|i| {
+                    OpenOptions::new()
+                        .read(true)
+                        .open(format!("{}/invalidated_ids.bin.{}", self.base_directory, i))
+                        .ok()
+                })
+                .collect(),
```

Comment:
do you need to sort? code is probably correct without the sort, but let's just sort it so in case we reuse this iterator for something else, it's not a trap. I think it'll be natural for others to assume we iterate on the order of insertion into this storage, but it's not the case here.